hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/StorageContainerLocationProtocolServerSideTranslatorPB.java

```diff
@@ -611,6 +638,28 @@ public ReplicationManagerStatusResponseProto getReplicationManagerStatus(
         .setIsRunning(impl.getReplicationManagerStatus()).build();
   }
 
+  public StartContainerBalancerResponseProto startContainerBalancer(
+      StartContainerBalancerRequestProto request)
+      throws IOException {
+    impl.startContainerBalancer();
+    return StartContainerBalancerResponseProto.newBuilder().build();
+  }
```

Comment:
Can we pass the parameters in the request to `ContainerBalancerConfiguration`?

Reply:
What Siddhant is suggesting is that before calling ContainerBalancer#start, a ContainerBalancerConfiguration can be generated from the input parameters. If user passes parameter of threshold as 0.05, it can be set in the configuration object and then the conf object can be passed to balancer. No other change would be required in Balancer.
@siddhantsangwan Please correct me if I am wrong.

Reply:
Yes, I think this will make managing configs easier.

Reply:
i see the conversion [here](https://github.com/apache/ozone/pull/2230#discussion_r639555281), which mentioned that balance configuration will not be needed. should we add it back？ maybe we could keep it like this for now , and add it back if needed as we move forward with the implementation. what do you think @lokeshj1703 @siddhantsangwan

Reply:
@JacksonYao287 ContainerBalancerConfiguration configuration would be needed. That is where all balancer related ozone configs are added. We should definitely add it back.
I think the conversation in #2230 was around having configuration as part of ContainerBalancer#start function. As @siddhantsangwan mentioned it would be good to have it there.

Reply:
thanks ,  it seems good , i will add it back , as well as some tests

Reply:
updated! PATL @lokeshj1703 @siddhantsangwan @GlenGeng